python/spear/sp_engine.py

```diff
@@ -0,0 +1,271 @@
+#
+# Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+#
+
+import msgpackrpc
+import os
+import psutil
+import spear
+from subprocess import Popen
+import sys
+import time
+
+class SpEngine():
+    def __init__(self, config):
+
+        self._config = config
+        self._rpc_client: msgpackrpc.Client
+
+        self._request_launch_unreal_instance()
+        self._initialize_rpc_client()
+
+        # Need to do these after we have a valid rpc_client
```

Comment:
The Python services should only need the RPC client, not the entire EngineService. In this sense, they are different (simpler and less tightly coupled) from the C++ services.

```
        self.engine_service = spear.EngineService(self._rpc_client)
        self.nav_mesh_service = spear.NavMeshService(self._rpc_client)
```